src/main/java/com/example/demo/service/MatchService.java

```diff
@@ -0,0 +1,70 @@
+package com.example.demo.service;
+
+import com.example.demo.models.Match;
+import com.example.demo.models.Team;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+@Component
+public class MatchService {
+    private List<Match> upcomingMatches = new ArrayList<>();
+    private List<Match> playedMatches = new ArrayList<>();
+
+     int a = 2;
+
+    private final TeamService teamService;
+
+    public MatchService(TeamService teamService) {
+        this.teamService = teamService;
+    }
+
+
+    public  Match playMatch() {
+        Match match = upcomingMatches.get(0);
+        String winnerName = new Random().nextBoolean()?match.getTeam1():match.getTeam2();
```

Comment:
I thing we will use this method many times. This service - bean - have singleton type. It means that we can create something here only once and use it many times. Instead of creating Random many times, you can do it only once and reuse it.